### ISSUE TITLE
`configure --prefix` no longer sets RUNPATH as expected

### ISSUE THREAD

**Issue by slingamn** [rakshasa/rtorrent#1166]
I'm on Ubuntu 22.04. I'm building libtorrent master (53596afc5fae27) via the following procedure:

1. `mkdir -p ~/sandbox`
2. `autoreconf -fi`
3. `./configure --prefix=/home/user/workspace/sandbox`
4. `make`
5. `make install`

then building rtorrent master (1e400144828c9) against it via the following procedure:

1. `export PKG_CONFIG_PATH=/home/user/workspace/sandbox/lib/pkgconfig`
2. `autoreconf -fi`
3. `./configure --prefix=/home/user/workspace/sandbox`
4. `make`
5. `make install`

Running `objdump -x` on the resulting `/home/user/workspace/sandbox/bin/rtorrent` reveals that it has no defined `RUNPATH` variable. Accordingly, it is unable to dynamically link against `/home/user/workspace/sandbox/lib/libtorrent.so.21` as required. A workaround is to `export LD_LIBRARY_PATH=/home/user/workspace/sandbox/lib`.

In contrast, when building the version of rtorrent immediately prior to #1102 , i.e. 52ceb39ca38a6, `RUNPATH` is set as expected:

```
Dynamic Section:
  NEEDED               libncursesw.so.6
  NEEDED               libtinfo.so.6
  NEEDED               libcurl.so.4
  NEEDED               libtorrent.so.21
  NEEDED               libstdc++.so.6
  NEEDED               libgcc_s.so.1
  NEEDED               libc.so.6
  RUNPATH              /home/user/workspace/sandbox/lib
```

so it seems like this was introduced by #1102.

Thanks very much for your time.

**Comment by slingamn** [comment]
This is no longer an issue as of ee120730cc046a9628bcd5d26f8a1f9b99fab158, but I'm not sure what fixed it.

**Comment by slingamn** [comment]
For future reference, it seems this was fixed by c1d076f1822752d4956a884a9aa9e6b0026b11f3.
